packages/ethereum-contracts/contracts/utils/DMZForwarder.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: AGPLv3
+pragma solidity 0.8.23;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title DMZForwarder
+ * @dev The purpose of this contract is to make arbitrary contract calls batchable
+ * alongside Superfluid specific batch operations.
+ * We route the calls through this dedicated contract in order to not have msg.sender set
+ * to the host contract, for security reasons.
+ * Forwarded calls can optionally use ERC-2771 to preserve the original msg.sender.
+ * If native tokens (msg.value) are provided, they are forwarded as well.
+ */
+contract DMZForwarder is Ownable {
+    /**
+     * @dev Forwards a call for which msg.sender doesn't matter
+     * @param target The target contract to call
+     * @param data The call data
+     */
+    function forwardCall(address target, bytes memory data)
+        external payable
+        returns(bool success, bytes memory returnData)
+    {
+        // solhint-disable-next-line avoid-low-level-calls
+        (success, returnData) = target.call{value: msg.value}(data);
```

Comment:
Is there any merit to using `address(this).balance` here or trying to return the leftover? (Probably not.)

Reply:
> Is there any merit to using address(this).balance here

I don't think so

> trying to return the leftover

No, but I added a function for withdrawing stuck native tokens.